src/ChannelBase.php

```diff
@@ -2,20 +2,127 @@
 
 namespace Drupal\share_light;
 
+/**
+ * A base class for social media channels.
+ */
 abstract class ChannelBase implements ChannelInterface {
   protected $block;
   protected $options;
-  public function __construct($block, $options = array()) {
+
+  /**
+   * Constructor function.
+   *
+   * @param Block $block
+   *   The corresponding `share_light` `Block`.
+   * @param array $options
+   *   An array of default values.
+   */
+  public function __construct(Block $block, array $options = array()) {
     $this->block = $block;
     $this->options = $options + static::defaults();
   }
+
+  /**
+   * Returns the default values for the channel.
+   */
   public static function defaults() {
     return array('toggle' => 1);
   }
+
+  /**
+   * Adds a widget containing the options for the channel.
```

Comment:
How about “Generate form for editing the channel specific settings.” or something similar?